File: Core/Entities/Receptie.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Core.Entities
{
    public class Receptie : BaseEntity
    {
        public int NrNir { get; set; }
        public DateTime DataNir { get; set; }
        public string NrDoc { get; set; }
        public DateTime DataDoc { get; set; }
        public string Obs { get; set; }
        public decimal BazaAch { get; set; }
        public decimal TvaAch { get; set; }
        public decimal ValAch { get; set; }
        public byte AdaosProc { get; set; }
        public decimal Baza { get; set; }
        public decimal Tva { get; set; }
        public decimal Val { get; set; }
        public int FirmaId { get; set; }
        public Firma Firma { get; set; }
        public int FurnizorId { get; set; }
        public Furnizor Furnizor { get; set; }
        public int TipDocumentId { get; set; }
        public  TipDocument TipDocument { get; set; }
        public ICollection<Material> Materiale { get; set; }

    }
}

File: client/src/app/shared/models/receptie.ts
export interface IMaterial {
    id: number;
    nrPozDoc: number;
    den: string;
    cant: number;
    um: string;
    cotaTvaAch: number;
    pretAch: number;
    bazaAch: number;
    tvaAch: number;
    valAch: number;
    cotaTva: number;
    adaosProc: number;
    pret: number;
    baza: number;
    tva: number;
    val: number;
    cantUtilizata: number;
    cantRamasa: number;
    receptieId: number;
}

export interface IReceptieDtoToSave {
    nrNir: number;
    dataNir: Date;
    tipDocumentId: number;
    nrDoc: string;
    dataDoc: Date;
    obs: string;
    bazaAch: number;
    tvaAch: number;
    valAch: number;
    adaosProc: number;
    baza: number;
    tva: number;
    val: number;
    firmaId: number;
    furnizorId: number;
    materiale: IMaterial[];
}


export interface IReceptieDto {
    id: number;
    nrNir: number;
    dataNir: Date;
    tipDocumentId: number;
    nrDoc: string;
    dataDoc: Date;
    obs: string;
    bazaAch: number;
    tvaAch: number;
    valAch: number;
    adaosProc: number;
    baza: number;
    tva: number;
    val: number;
    firmaId: number;
    furnizorId: number;
    furnizor: string;
    materiale: IMaterial[];
}

export interface IReceptieInListDto {
    id: number;
    nrNir: number;
    dataNir: Date;
    furnizor: string;
    tipDocument: string;
    nrDoc: string;
    dataDoc: Date;
    bazaAch: number;
    tvaAch: number;
    valAch: number;
}

export interface IPaginationReceptie {
    pageIndex: number;
    pageSize: number;
    count: number;
    data: IReceptieInListDto[];
}

export class ReceptieParams {
    pageIndex = 1;
    pageSize = 10;
    nrDoc: string;
    furnizorId: number;
}


